### ISSUE TITLE
Improve user experience for Excel files

### ISSUE THREAD

**Issue by suncloudsmoon** [open-webui/open-webui#2867]
**Is your feature request related to a problem? Please describe.**
When I am trying to ask a chatbot about the details of a certain Excel file, the whole excel file is fed into the RAG context without having the ability to specify specific rows/columns (in the UI) to narrow down the focus.

**Describe the solution you'd like**
Display the Excel file side by side with the chat interface.

**Describe alternatives you've considered**
We could specify the column or row by some special identifier like "[row,col]" to chatbot. However, it isn't that user friendly.

**Additional context**
[How ChatGPT currently handles Excel files.](https://openai.com/index/improvements-to-data-analysis-in-chatgpt/)

**Comment by CellCS** [comment]
@suncloudsmoon  may i ask which embedding model  you use for excel or csv file? I always get wrong row number from response. thanks

**Comment by pbpippen** [comment]
Agree. Having trouble working with excel files. Either doesn't get context correctly or only partial info is returned

**Comment by suncloudsmoon** [comment]
> @suncloudsmoon may i ask which embedding model you use for excel or csv file? I always get wrong row number from response. thanks
I'm not using any embedding models right now. I tend to avoid asking LLMs about specific data as it typically hallucinates for certain models.

**Comment by BetDev1** [comment]
@suncloudsmoon  Thanks, I have tried nomic-embed-text from ollama and text-embedding-3-small from OpenAI. But I think model is not issue here, because if using AnythingLM, the knowledge-based chat are more precious with using the same embedding models and LLM.  You also could make local try by using AnythingLM + ollama/LM_Studio. But AnythingLM is desktop app, I need to use OpenWebUI for web app.

**Comment by CellCS** [comment]
@suncloudsmoon  thanks, i will do a comparation, then update.
